### ISSUE TITLE
Can not get debug to work. requires Bash to be the default shell. How to configure this.

### ISSUE THREAD

**Issue by dbridwell1** [codefori/vscode-ibmi#2044]
I am starting the debug server but an error pops up saying "Bash is installed on the IBM i, but it is not your default shell. Please switch to bash to setup the debug service.  
However, as far as I know Bash is my default. 
When I ssh into the system it comes up as bash.
Or if I hover over my system name, click terminals and select PASE it comes up as bash
bash-5.2 if that makes a difference. 
However, if I add a new terminal by clicking on the + it comes up as Powershell. 
I can not find the setting for this default shell.

**Comment by sebjulliand** [comment]
This message appears if the output of `echo $SHELL` does not return `/QOpenSys/pkgs/bin/bash`.
The default shell cannot be found in VSCode itself, but running `echo $SHELL` in a Terminal opened after hovering over your connection name will be just fine.

More information about setting Bash as your default shell can be found here: https://ibmi-oss-docs.readthedocs.io/en/latest/troubleshooting/SETTING_BASH.html

**Comment by dbridwell1** [comment]
I did have my user profile set to use bash and running echo $SHELL did return the correct path.   Then I tried changing the Default shell to use bash.  Still didn't work.  Then I changed my user profile to use the default shell by setting it to NULL. Then it started working.  Got the debug server to start.  Now it says the debug service is using referencing an existing web server.  I think it is because that server is using the default port 8001 and 8005.  Now I need to find info on changing the ports.

**Comment by SJLennon** [comment]
@dbridwell1 
Info is in here at "Question: How can I change the port numbers for the debug service?"
https://www.ibm.com/docs/en/merlin/1.0?topic=environment-debugging

**Comment by dbridwell1** [comment]
I did get it working!!!  Found the port settings in the debug settings page.  Changed those to some unused ports (8101 and 8105) then restarted and it worked!  Thanks for the assistance.

**Comment by SJLennon** [comment]
@dbridwell1 
And see "Configuring & starting the service" under Debug in the documentation  (https://codefori.github.io/docs/developing/debug/)

**Comment by SJLennon** [comment]
@dbridwell1 
> Then I changed my user profile to use the default shell by setting it to NULL.
Curiosity question:  How did you do this?  Something thing like this, maybe: `CALL QSYS2.SET_PASE_SHELL_INFO('*CURRENT', '')`

(And you may want to close this issue since it appears you have everything fixed 😃 )

**Comment by dbridwell1** [comment]
Yes - "I used call qsys2.set_pase_shell_info('*CURRENT', '')"
then "call qsys2.set_pase_shell_info('*DEFAULT', '/QOpenSys/pkgs/bin/bash')"
I think it was more because of the port problem.  When the debug service started it was pointing at a job that was already started and running under a different user id. I have Fortra's GoAnywhere running on that port. I saw another troubleshooting tip where someone else had that software running on the same port. I think because the job was already running under a different user id that that user id needed to be set to bash.  I think changing the shell default probably allowed that.  Even though that error went away the job showed up  as the GoAnywhere job.  Then I changed the ports and everything worked fine. 
Thanks for the help!
